rasa_sdk/tracing/trace_provider.py

```diff
@@ -0,0 +1,23 @@
+from typing import Optional
+from rasa_sdk.utils import Singleton
+from opentelemetry.trace import Tracer
+
+
+class TraceProvider(metaclass=Singleton):
```

Comment:
I would rename this component not to be confused with OpenTelemetry's TracerProvider.
Let's rename it to `TracerRegister` or something like that.